osquery/tables/system/darwin/xprotect.cpp

```diff
@@ -147,72 +150,86 @@ QueryData genXProtectReports(QueryContext& context) {
 QueryData genXProtectEntries(QueryContext& context) {
   QueryData results;
   pt::ptree tree;
+  bool plistFound = false;
```

Comment:
(Maybe this is a style thing, and I don't know what we'd prefer in c++) Is this boolean needed?

I'd probably just loop, and when parsed successfully, early return. Falling out of the loop implicitly means nothing was found.

Reply:
I guess maybe @theopolis was encouraging this approach.

Reply:
Probably depends a bit on whether we'd ever read _both_. If we stop after the first successful read, I'd advocate early return and no boolean. If we read them all, then the boolean is needed

Reply:
Yea I was generally following the recommendation of @theopolis . I think that with the approach of checking all the potential plists, it make sense to try read each one, but stopping after a successful read wouldn't be bad.

Reply:
I don't know which usage mode is correct here.

Reply:
@jschroth mentioned in https://github.com/osquery/osquery/pull/7142#issuecomment-856101597 that he has seen a macOS 11.1 with the old path, I'm not sure if files were there in both the new and old directory or not.
Seems safest to try both files.

Reply:
Order probably matters. On the theory that there might be old stale entries, try the new one first?

Reply:
that makes sense, I'll update it.

Reply:
For context, the 11.1 instance only has the old location (i.e. it was not in the `/Library/Apple` dir so osquery is working there) and I agree with @directionless on doing an early return once it is found.

I also have an 11.3 and and an 11.4 instance and I first noticed the change in my 11.3 instance that I created ~1mo ago and found the XProtect was not working just this last week while I was doing some testing which is why I created #7142.  I realized it was already being discussed here and so I abandoned my PR. ;-)

Reply:
Ah ok, it seems that we're unlikely to encounter 2 locations plists then. I've updated the code so to use early returns to indicate success instead of tracking whether a valid plist was found.